apps/dashboard/app/models/project.rb

```diff
@@ -80,61 +78,70 @@ def templates
   attr_accessor :template
 
   def initialize(attributes = {})
-    @id = attributes.delete(:id)
-    @directory = attributes.delete(:directory)
+    @id = attributes.fetch(:id, Project.next_id)
+    update_attrs(attributes)
+    @directory = attributes[:directory]
     @directory = File.expand_path(@directory) unless @directory.blank?
-    @template = attributes.delete(:template)
+    @template = attributes[:template]
 
-    @manifest = Manifest.new(attributes)
-    @manifest = @manifest.merge(Manifest.load(manifest_path)) unless new_record?
-  end
+    return if new_record?
 
-  def new_record?
-    return true if !id
-    return true if !directory
+    contents = File.read(manifest_path)
+    raw_opts = YAML.safe_load(contents)
+    update_attrs(raw_opts.symbolize_keys)
+  end
```

Comment:
I just realized that `id` is never read from the manifest file.
I am wondering if we should load the manifest into a hash first before calling `initialize` on `Project.all` and `Project.find`. So that the `id` in the Project is the one in the manifest instead of the one in the `.project_lookup`

Reply:
Yea I was thinking about the same. I think that'll come into play more when we have to discover other projects by searching directories.  I'd say let's defer it until we build that use case.

I realize it's all a bit in flux and subject to change, I think we just need to be smart at each iteration so we don't build a lot of cruft.